src/main/java/nl/workingtalent/wtacademy/reservation/ReservationController.java

```diff
@@ -1,11 +1,140 @@
 public class ReservationController {
 
 	@Autowired
 	private ReservationService service;
+
+	@Autowired
+	private BookService bookService;
+
+	@Autowired
+	private UserService userService;
+
+	// READ
+	@RequestMapping("reservation/all")
+	public ResponseDto findAllReservations() {
+		List<Reservation> reservations = service.findAllReservations();
+
+		if (reservations.isEmpty()) {
+			return new ResponseDto(false, null, null, "No reservations found.");
+		}
+
+		Stream<ReadReservationDto> readReservationDtoStream = getReservations(reservations);
+		return new ResponseDto(true, readReservationDtoStream, null,
+				reservations.size() + (reservations.size() < 2 ? " reservation " : " reservations ") + "found.");
+	}
+
+	@RequestMapping("reservation/{id}")
+	public ResponseDto findReservationById(@PathVariable("id") long id) {
+		Optional<Reservation> reservationOptional = service.findReservationById(id);
+
+		if (reservationOptional.isPresent()) {
+			Reservation reservation = reservationOptional.get();
+			ReadReservationDto readReservationDto = new ReadReservationDto(reservation);
+			return new ResponseDto(true, readReservationDto, null, "Reservation found.");
+		}
+		return new ResponseDto(false, null, null, "No reservation found.");
+	}
+
+	@RequestMapping("reservation/search")
+	public ResponseDto searchReservation(@RequestBody SearchReservationDto dto) {
+		List<Reservation> reservations = service.searchReservations(dto);
+		Stream<ReadReservationDto> dtos = reservations.stream().map((reservation) -> {
+			return new ReadReservationDto(reservation);
+		});
+		return new ResponseDto(true, dtos, null,
+				reservations.size() + (reservations.size() < 2 ? " reservation " : " reservations ") + "found.");
+	}
+
+	// CREATE
+	@PostMapping("reservation/create")
+	public ResponseDto createReservation(@RequestBody CreateReservationDto dto) {
+		Optional<Book> optionalBook = bookService.getBookById(dto.getBook());
+
+		// DOES BOOK EXIST?
+		if (optionalBook.isPresent()) {
+			Book book = optionalBook.get();
+			
+			// DOES USER EXIST?
+			Optional<User> optionalUser = userService.findUserById(dto.getUser());
+			if (optionalUser.isPresent()) {
+				User user = optionalUser.get();
+
+				Reservation newReservation = new Reservation();
+				newReservation.setReservationRequest(dto.getReservationRequest());
+				newReservation.setRequestDate(dto.getRequestDate());
+				newReservation.setBook(book);
+				newReservation.setUser(user);
+				service.create(newReservation);
+				return new ResponseDto(true, newReservation.getId(), null, "Reservation created successfully.");
+			} else {
+				return new ResponseDto(false, null, null, "User not found.");
+			}
+		} else {
+			return new ResponseDto(false, null, null, "Book not found.");
+		}
+	}
+
+	// UPDATE
+	@PutMapping("reservation/update/{id}")
+	public ResponseDto updateReservation(@RequestBody UpdateReservationDto dto, @PathVariable("id") long id) {
+
+		// DOES RESERVATION EXIST?
```

Comment:
In plaats van via pathvariable, kan je de id gewoon meegeven binnen de updateDto.

Reply:
Fixed.